### DISCUSSION TITLE
Difference between SimpleTrigger.FireNow & IgnoreMisfirePolicy in one-shot jobs

### DISCUSSION THREAD

**Discussion by giacomodeliberali** [quartznet/quartznet#1076] [Q&A] [answered]
In a one-shot trigger with a `SimpleSchedule`, what's the difference of using `MisfireInstruction.IgnoreMisfirePolicy` and `MisfireInstruction.SimpleTrigger.FireNow`?

When catching a misfire (on app startup for example) both policies are gonna fire that trigger (now), aren't they?

**Comment by lahma** [comment, marked as answer]
If I remember correctly from the top of my head... in this case, say you have trigger that has scheduled start at `2021-01-14 09:00` and interval of 1 hour. When you start your software at `2021-01-14 11:15` with ignore it should run next time `2021-01-14 12:00` and with immediately it would fire the time software starts (trigger misfired for 09:00). For simple trigger the trigger's start time is big part of the puzzle. And of course persistence, if it's RAMJobStore based logic starts from the assumption that as history is missing there's a misfire if the start time is in the past.

I reserve the right to remember wrong the details 😉

**Reply by giacomodeliberali** [reply]
Suppose my job is scheduled to run a single time at  `2021-01-14 09:00` (no interval, just start date and zero repeat count with sql persistence). What would happen when I start my app at `2021-01-14 15:00`  with the different policies? If I have understood properly both of them should fire and execute the job as soon as they can (so around `2021-01-14 15:00`). It that correct?

**Reply by lahma** [reply]
Sounds about right. With database persistence the trigger is removed after the fire as it's considered never firing again so if you always schedule on start the trigger should reappear and be considered as misfired and should fire as soon as possible.
